dsgrid/project.py

```diff
@@ -226,8 +226,37 @@ def process_query(self, context: QueryContext, cached_datasets_dir: Path):
             datasets[dataset_id] = DatasetExpressionHandler(
                 df.select(*expected_columns), time_columns + dim_columns, pivoted_columns_sorted
             )
+        return datasets
 
-        return evaluate_expression(context.model.project.dataset.expression, datasets).df
+    def _convert_unpivoted_datasets(self, context: QueryContext, filenames: dict[str, Path]):
+        dim_columns, time_columns = self._get_dimension_columns(context)
+        expected_columns = time_columns + dim_columns
+        expected_columns.append(VALUE_COLUMN)
+
+        datasets = {}
+        for dataset_id, path in filenames.items():
+            df = read_dataframe(path)
+            unexpected = sorted(set(df.columns).difference(expected_columns))
+            if unexpected:
+                raise Exception(f"Unexpected columns are present in {dataset_id=} {unexpected=}")
+            datasets[dataset_id] = DatasetExpressionHandler(
+                df.select(*expected_columns), time_columns + dim_columns, [VALUE_COLUMN]
```

Comment:
I guess `df.select(*expected_columns)` here makes sure all expected_columns are present in the df. Nothing to do

Reply:
It sets the column order.